HopsanGUI/loadFunctions.cpp

```diff
@@ -230,6 +230,16 @@ void loadParameterValue(QDomElement &rDomElement, ModelObject* pObject, UndoStat
         return;
     }
 
+    if(existingNames.contains(parameterValue) &&
+       !pObject->getParentContainerObject()->hasParameter(parameterValue))
+    {
+        gpMessageHandler->addWarningMessage(QString("%1: Updating parameter value from \"%2\" to \"self.%2\"")
+                                            .arg(pObject->getName())
+                                            .arg(parameterValue));
+        pObject->setParameterValue(parameterName, "self."+parameterValue, true);
+        return;
+    }
+
```

Comment:
This code will not update NumHop expressions containing parameter names. Only if the value is exactly a parameter name.

Also this code makes the model loose backwards compatibility. You wont be able to load it in an older version of Hopsan (unless you manually remove the self.). But I know this is quit difficult to solve with a reasonable amount of work.

The same update must be made in the Core HMF loader otherwise the models will fail when run through the CLI and remote. (TEsts are currently failing) likely this is the cause. (Not sure)

Reply:
I think updating NumHop scripts would be really complicated. Or is there some function for easily replacing symbol names?

Loading new models with old versions of Hopsan is never guaranteed to work, and I see no solution for this really. It would require a "Hopsan version" setting in the save dialog, to save in older HMF formats. Old Hopsan versions of course have no idea about the meaning of "self."...

Reply:
If the parameter value is a script, just go through the text and replace parName with self.parName. Look for an occurance of any local parameter name.

Reply:
@peterNordin Will that really work? Consider the case that the expression is "x+xp", where x is a local parameter and xp is a system parameter. The correct replacement will be "self.x+xp". But since it will replace all "x" with "self.x", it will end up with "self.x+self.xp". It would require a NumHop function that will rename all occurrences of a variable.

Reply:
Yes, sure we would need to parse the expression and extract variable names and check which are not system parameters.  Maybe its not worth the effort really.